### PR TITLE
Update bamit to return vector of records.

### PR DESCRIPTION
**Pull request by joshuak94**

### CODE REVIEW

File: test/api/search_test.cpp [hunk 1]
```diff
@@ -40,7 +39,7 @@ TEST(get_overlap_records, simulated_chr1_small_golden)
     // Make sure both files are at the end of their results.
     EXPECT_EQ(it_result, result.end());
     EXPECT_EQ(it_expected, expected.end());
-    
-     std::filesystem::remove(result_sam_path);
+
+     // std::filesystem::remove(result_sam_path);
```
Comment (MitraDarja) [inline test/api/search_test.cpp:43]: Why is this commented out?

**Review by MitraDarja — COMMENTED** [review]
LGTM

File: test/api/search_test.cpp (hunk 1 above)
Comment (joshuak94) [reply test/api/search_test.cpp:43]: Oops :D

**Comment by joshuak94** [comment]
@Irallia The goal is to have a function which returns a "seqan-like" vector of sam_record objects, so that I could do something like this:
```cpp
auto records_vector = bamit::get_overlap_records(input, index, start, end)
uint32_t avg_read_depth = records_vector.size() / (end - start);
```

At the moment I was only returning the file position, so the user would have to parse the records themself. This just makes it a bit easier.

**Review by Irallia — APPROVED** [review]
There is relatively much functionality to follow. I have not evaluated every step but what I have understood looks very good. If you want me to look over any part in more detail, let me know.
I have comments only to the documentation.

**Pull request merged**